### PR TITLE
feat(quest-page): add auto expand and tab filters

### PR DESCRIPTION
**Pull request by eliobricenov**
### Summary

closes https://github.com/HyperPlay-Gaming/product-management/issues/627

### How to test:

- go to the quest page
image
- check quest detail is auto expanded 
image

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: src/frontend/screens/Quests/components/QuestsSummaryTableWrapper/index.tsx
Comment (eliobricenov) [inline src/frontend/screens/Quests/components/QuestsSummaryTableWrapper/index.tsx:1]: The changes are basically moving the filter/sort state from this component to one level above. Once the filtering is the parent component, we can default to the first quest **after** filtering/sorting and pass the selected quest id to the quest list and quest detail components.

Comment (eliobricenov) [reply src/frontend/screens/Quests/components/QuestsSummaryTableWrapper/index.tsx:1]: After moving the state, the wrapper component is pretty much empty so I decided to remove it and implement QuestSummaryTable directly in the quest page component.

**Review by jiyuu-jin — APPROVED** [review]
Worked and code LGTM

image

**Review by BrettCleary — APPROVED** [review]
LGTM. There is a short period on the initial load where the quest details wrapper is not shown, then some layout shift once the quests are fetched. It would be nice if the quest details component was in a loading state at this point too to prevent layout shift.

Here is what it looks like
image


There also seems to be a regression in the quest details background color and border either on this PR or a prior PR

On this PR:
image

On figma: https://www.figma.com/design/yHrBPoE70k1I2icW4paEit/Quests?node-id=1-17948&t=tBaRuRLtOpcveXar-4

**Review by nyghtstalker — APPROVED** [review]
Quests are now always auto-expanded by default. 

image

**Pull request merged**